nixos/modules/services/x11/xserver.nix

```diff
@@ -552,6 +561,9 @@ in
                 + "${toString (length primaryHeads)} heads set to primary: "
                 + concatMapStringsSep ", " (x: x.output) primaryHeads;
       })
+      { assertion = cfg.desktopManager.default == "none" && cfg.windowManager.default == "none" -> cfg.plainX;
+        message = "Either the desktop manager or the window manager shouldn't be `none`!";
```

Comment:
The best error message is the error message which let you fix the issue which is being reported.
This message should include some hint to fix the configuration.

Reply:
agreed. Added another sentence which mentions the `plainX` option ...